relay-general/src/protocol/measurements.rs

```diff
@@ -0,0 +1,207 @@
+use crate::types::{Annotated, Error, FromValue, Object, Value};
+
+/// An individual observed measurement.
+#[derive(Clone, Debug, Default, PartialEq, Empty, FromValue, ToValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct Measurement {
+    /// Value of observed measurement value.
+    #[metastructure(required = "true")]
+    pub value: Annotated<f64>,
+}
+
+/// A map of observed measurement values.
+///
+/// Measurements are only available on transactions. They contain measurement values of observed
+/// values such as Largest Contentful Paint (LCP).
+#[derive(Clone, Debug, Default, PartialEq, Empty, ToValue, ProcessValue)]
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct Measurements(pub Object<Measurement>);
+
+impl FromValue for Measurements {
+    fn from_value(value: Annotated<Value>) -> Annotated<Self> {
+        let mut processing_errors = Vec::new();
+
+        let mut measurements = Object::<Measurement>::from_value(value).map_value(|measurements| {
+            let measurements = measurements
+                .into_iter()
+                .filter_map(|(name, value)| {
+                    let name = name.trim();
+                    if let Some(measurement_value) = value.value() {
+                        if is_valid_measurement_name(name) {
+                            if measurement_value.value.value().is_some() {
+                                return Some((name.to_lowercase(), value));
+                            } else {
+                                let new_value = Measurement {
+                                    value: Annotated::from_error(
+                                        Error::invalid("measurement value must be numeric"),
```

Comment:
please check how numeric values are asserted for otherwise, I think `Error::expected("a numeric value")`

Reply:
My understanding is that this is already done by the derived from value implementation. I think you simply have to leave the deserialized measurement in the map, and you'll see an error.